File: desh/ege2026kp/24solve/24-177.pas
// Автор В.Н. Шубинкин
##
assign(input, '24-157.txt');
var s := Readlnstring;
var k := 1;  // k - позиция начала текущей подходящей цепочки символов
var maxLen := 0;
for var i := 2 to s.Length do
begin
  if (s[i - 1] + s[i] = 'PR') or (s[i - 1] + s[i] = 'RP') then
  begin
    if i - k > maxLen then 
      maxLen := i - k;
    k := i;
  end;  
end;
print(maxLen); 

File: desh/ege2026kp/24solve/24-177.py
with open('24-157.txt') as F:
  s = F.readline()

#s = 'RPASPRLKJRPQWERPR'

maxLen = 0
curLen = 0
prev = '*'
for c in s:
  if (prev == 'P' and c == 'R') or \
     (prev == 'R' and c == 'P'):
    curLen = 1
  else:
    curLen += 1
    maxLen = max( maxLen, curLen )
  #print(prev, c, curLen)
  prev = c

print( maxLen )
